---
categories:
 - weekly
title: "13 - Differential expression"
slides: ""
tags:
 - assets
visible: 1
---


[Slides: Introduction into differential expression analysis]({{site.baseurl}}/presentations/07a_DiffExpr/07a_Intro.pdf)  
[Exercise: `lab/DEG_ttest.R` - Basic t-test and volcano plot](https://github.com/mdozmorov/BIOS567.2017/tree/gh-pages/assets/07a_DiffExpr/lab)  
[Exercise: `lab/volcano.R` - Demo of pretty volcano plot](https://github.com/mdozmorov/BIOS567.2017/tree/gh-pages/assets/07a_DiffExpr/lab)  
[Handouts: Power analysis]({{site.baseurl}}/presentations/07a_DiffExpr/Power_analysis.pdf)  
[Slides: Linear models for microarray data analysis, limma, empirical Bayes]({{site.baseurl}}/presentations/07a_DiffExpr/07b_limma.pdf)  
[Exercise: lab/DiffExpr_Limma.Rmd - limma demo](https://github.com/mdozmorov/BIOS567.2017/tree/gh-pages/assets/)  
[Slides: Significance Analysis of Microarrays, SAM]({{site.baseurl}}/presentations/07a_DiffExpr/07c_SAM.pdf)  
[Exercise: lab/DiffExpr_SAM.Rmd - SAM demo, uses GDS858 data. BiomaRt demo at the end](https://github.com/mdozmorov/BIOS567.2017/tree/gh-pages/assets/)  
[Slides: Multiple testing correction]({{site.baseurl}}/presentations/07a_DiffExpr/07d_multipletesting.pdf)  
[Exercise: lab/Filtering.Rmd - filtering gene set](https://github.com/mdozmorov/BIOS567.2017/tree/gh-pages/assets/)  
[Exercise: lab/multitest.R - multiple testing correction and filtering. Uses Methyl.RData](https://github.com/mdozmorov/BIOS567.2017/tree/gh-pages/assets/)  
[Slides: Batch effect correction, ComBat, SVA]({{site.baseurl}}/presentations/07a_DiffExpr/07e_batch.pdf)  
[Batch effect exercise 1](https://genomicsclass.github.io/book/pages/adjusting_with_linear_models.html), and [exercise 2](http://jtleek.com/genstats/inst/doc/02_13_batch-effects.html)  





## References

- Krzywinski, Martin, and Naomi Altman. “Points of Significance: Comparing Samples—part I.” Nature Methods 11, no. 3 (March 2014): 215–16.
- Krzywinski, Martin, and Naomi Altman. “Points of Significance: Analysis of Variance and Blocking.” Nature Methods 11, no. 7 (July 2014): 699–700.
- Krzywinski, Martin, and Naomi Altman. “Points of Significance: Power and Sample Size.” Nature Methods 10, no. 12 (November 26, 2013): 1139–40. doi:10.1038/nmeth.2738.

- Cui, Xiangqin, and Gary A. Churchill. “Statistical Tests for Differential Expression in CDNA Microarray Experiments.” Genome Biology 4, no. 4 (2003): 210. - Differential expression analysis of microarrays, from fold-change to t-test, its moderated versions SAM and limma, and ANOVA. 

- Tong, Tiejun, and Hongyu Zhao. “Practical Guidelines for Assessing Power and False Discovery Rate for a Fixed Sample Size in Microarray Experiments.” Statistics in Medicine 27, no. 11 (May 20, 2008): 1960–72. doi:10.1002/sim.3237. Power analysis. t-statistics, FDR types and definitions, then derivation of power calculations. 

- Video "Statistics for Genomics: Advanced Differential Expression" by Rafael Irizarry - limma statistics, Bayesian intro [https://www.youtube.com/watch?v=QINX3cI7qgk](https://www.youtube.com/watch?v=QINX3cI7qgk)

### Limma

- Lönnstedt, Ingrid, and Terry Speed. “REPLICATED MICROARRAY DATA.” Statistica Sinica 12, no. 1 (2002): 31–46. http://www.jstor.org/stable/24307034. - Empirical Bayes method for analyzing microarray replicates. Issues with simple approaches, proposed B statistics - a Bayes log posterior odds with two hyperparameters in the inverse gamma prior for the variances, and a hyperparameter in the normal prior of the nonzero means. Appendix - detailed definitions, derivations, and solutions.

- Smyth, Gordon K. “Linear Models and Empirical Bayes Methods for Assessing Differential Expression in Microarray Experiments.” Statistical Applications in Genetics and Molecular Biology 3 (2004): Article3. doi:10.2202/1544-6115.1027. [PDF](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.315.2066&rep=rep1&type=pdf) - Linear models for differential analysis, moderated t-statistics via shrinkage of sample variance. Empirical estimation of Bayesian prior variance distribution and shrinkage hyperparameters.

- Phipson, Belinda, Stanley Lee, Ian J. Majewski, Warren S. Alexander, and Gordon K. Smyth. “Robust Hyperparameter Estimation Protects against Hypervariable Genes and Improves Power to Detect Differential Expression.” The Annals of Applied Statistics 10, no. 2 (June 2016): 946–63. doi:10.1214/16-AOAS920. [PDF](https://projecteuclid.org/download/pdfview_1/euclid.aoas/1469199900) - An extension of differential analysis using linear modeling and empirical Bayes by windsorizing outliers in estimating sample distribution. 

#### Extensions of Limma method

- Sartor, Maureen A., Craig R. Tomlinson, Scott C. Wesselkamper, Siva Sivaganesan, George D. Leikauf, and Mario Medvedovic. “Intensity-Based Hierarchical Bayes Method Improves Testing for Differentially Expressed Genes in Microarray Experiments.” BMC Bioinformatics 7 (December 19, 2006): 538. https://doi.org/10.1186/1471-2105-7-538. https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-7-538 - Intensity-Based Moderated T-statistic (IBMT). Empirical Bayes approach allowing for the relationship between variance and gene signal intensity (estimated with loess). Brief description of previous methods (Smyth, Cyber-T). Details of Smyth hierarchical model and moderated t-statistics, estimation of hyperparameters with implementation of variance-signal. Software at http://eh3.uc.edu/ibmt/. 

- Lianbo Yu et al., “Fully Moderated T-Statistic for Small Sample Size Gene Expression Arrays,” Statistical Applications in Genetics and Molecular Biology 10, no. 1 (September 15, 2011), https://doi.org/10.2202/1544-6115.1701. https://www.degruyter.com/view/j/sagmb.2011.10.issue-1/1544-6115.1701/1544-6115.1701.xml - Third implementation of moderated t-statistics. First is Smyth 2004 model assuming $d_{0g}$ and $s_{0g}^2$ constant, second is IBMT (intensity-based moderated t) Sartor 2006 allows varying $s_{0g}^2$ with gene expression, third is the present FMT (fully moderated t) model allowing varying d_{0g} and s_{0g}^2. Description of Smyth hierarchical model, estimation of hyperparameters. Adjusted log variances are fit with loess. Goal - increase in power - is demonstrated on simulated data.

### SAM - Significance analysis of Microarrays

- Tusher, V. G., R. Tibshirani, and G. Chu. “Significance Analysis of Microarrays Applied to the Ionizing Radiation Response.” Proceedings of the National Academy of Sciences of the United States of America 98, no. 9 (April 24, 2001): 5116–21. https://doi.org/10.1073/pnas.091062498.

### Multiple testing 

- Goeman, Jelle J., and Aldo Solari. “Multiple Hypothesis Testing in Genomics.” Statistics in Medicine 33, no. 11 (May 20, 2014): 1946–78. https://doi.org/10.1002/sim.6082. - multiple testing review

- Benjamini, Yoav, and Yosef Hochberg. “Controlling the False Discovery Rate: A Practical and Powerful Approach to Multiple Testing.” Journal of the Royal Statistical Society. Series B (Methodological), 1995, 289–300. - FDR paper

- Storey, John D., and Robert Tibshirani. “Statistical Significance for Genomewide Studies.” Proceedings of the National Academy of Sciences of the United States of America 100, no. 16 (August 5, 2003): 9440–45. https://doi.org/10.1073/pnas.1530509100. - q-value paper

- Krzywinski, Martin, and Naomi Altman. “Points of Significance: Comparing Samples—part II.” Nature Methods 11, no. 4 (March 28, 2014): 355–56. doi:10.1038/nmeth.2900.

### Batch effect

- Lazar, Cosmin, Stijn Meganck, Jonatan Taminau, David Steenhoff, Alain Coletta, Colin Molter, David Y. Weiss-Solís, Robin Duque, Hugues Bersini, and Ann Nowé. “Batch Effect Removal Methods for Microarray Gene Expression Data Integration: A Survey.” Briefings in Bioinformatics 14, no. 4 (July 2013): 469–90. https://doi.org/10.1093/bib/bbs037. - Review of batch correction methods, definitions, methodologies

- Leek, Jeffrey T., Robert B. Scharpf, Héctor Corrada Bravo, David Simcha, Benjamin Langmead, W. Evan Johnson, Donald Geman, Keith Baggerly, and Rafael A. Irizarry. “Tackling the Widespread and Critical Impact of Batch Effects in High-Throughput Data.” Nature Reviews. Genetics 11, no. 10 (October 2010): 733–39. https://doi.org/10.1038/nrg2825. - Batch effect, types, sources, examples of wrong conclusions, SVA and ComBat methods

- Johnson, W. Evan, Cheng Li, and Ariel Rabinovic. “Adjusting Batch Effects in Microarray Expression Data Using Empirical Bayes Methods.” Biostatistics (Oxford, England) 8, no. 1 (January 2007): 118–27. https://doi.org/10.1093/biostatistics/kxj037. - ComBat paper. Batch effect removal using Empirical Bayes method.

- Leek, Jeffrey T., and John D. Storey. “Capturing Heterogeneity in Gene Expression Studies by Surrogate Variable Analysis.” PLoS Genetics 3, no. 9 (September 2007): 1724–35. https://doi.org/10.1371/journal.pgen.0030161. - SVA paper

- Chen, Chao, Kay Grennan, Judith Badner, Dandan Zhang, Elliot Gershon, Li Jin, and Chunyu Liu. “Removing Batch Effects in Analysis of Expression Microarray Data: An Evaluation of Six Batch Adjustment Methods.” PloS One 6, no. 2 (2011): e17238. https://doi.org/10.1371/journal.pone.0017238. - Comparing batch effect removal software. ComBat is best. Importance of standartization
