---
categories:
 - weekly
title: "15 - Clustering"
slides: ""
tags:
 - assets
visible: 1
---

[Slides: Introduction to clustering, preprocessing, distance/similarity metrics, hierarchical/divisive clustering, single/average/complete metrics]({{site.baseurl}}/presentations/09_Clustering/09_Clustering.pdf)  
[Exercises: `Average_linkage.R`, `Single_Linkage.R`, `Complete_Linkage.R`, `Divisive_Kmeans.R`, `CramersV_gower.R`](https://github.com/mdozmorov/BIOS567.2017/tree/gh-pages/assets/09_Clustering/lab)  
[Slides: Non-hierarchical Clustering and dimensionality reduction techniques]({{site.baseurl}}/presentations/09_Clustering/09_Clustering1.pdf)  



## References

- Patrik D’haeseleer, “How Does Gene Expression Clustering Work?,” Nature Biotechnology 23, no. 12 (December 2005): 1499–1501, https://doi.org/10.1038/nbt1205-1499. - Clustering distances. Recommendations for gene expression choices of clustering

- Satagopan, Jaya M., and Katherine S. Panageas. “A Statistical Perspective on Gene Expression Data Analysis.” Statistics in Medicine 22, no. 3 (February 15, 2003): 481–99. doi:10.1002/sim.1350. - Intro into microarray technology, statistical questions. Hierarchical clustering - clustering metrics. MDS algorithm. Class prediction - linear discriminant analysis algorithm and cross-validation. SAS and S examples

- Altman, Naomi, and Martin Krzywinski. “Points of Significance: Clustering.” Nature Methods 14, no. 6 (May 30, 2017): 545–46. doi:10.1038/nmeth.4299. - Clustering depends on gene scaling, clustering method, number of simulations in k-means clustering.

- Abdi, Hervé, and Lynne J. Williams. “Principal Component Analysis.” Wiley Interdisciplinary Reviews: Computational Statistics 2, no. 4 (July 2010): 433–59. doi:10.1002/wics.101. - PCA in-depth review. Mathematical formulations, terminology, examples, interpretation. Figures showing PC axes, rotations, projections, circle of correlation. Rules for selecting number of components. Rotation - varimax, promax, illustrated. Correspondence analysis for nominal variables, Multiple Factor Analysis for a set of observations described by several groups (tables) of variables. Appendices - eigenvalues and eigenvectors, positive semidefinite matrices, SVD

- Wall, Michael. “Singular Value Decomposition and Principal Component Analysis,” n.d. https://link.springer.com/chapter/10.1007/0-306-47815-3_5 - SVD and PCA statistical intro. Relation of SVD to PCA, Fourier transform. Examples of applications, including genomics. 

- Lever, Jake, Martin Krzywinski, and Naomi Altman. “Points of Significance: Principal Component Analysis.” Nature Methods 14, no. 7 (June 29, 2017): 641–42. doi:10.1038/nmeth.4346. PCA explanation, the effect of scale. Limitations

- Meng, Chen, Oana A. Zeleznik, Gerhard G. Thallinger, Bernhard Kuster, Amin M. Gholami, and Aedín C. Culhane. “Dimension Reduction Techniques for the Integrative Analysis of Multi-Omics Data.” Briefings in Bioinformatics 17, no. 4 (July 2016): 628–41. doi:10.1093/bib/bbv108. - A must. Dimensionality reduction techniques - PCA and its derivatives, NMF. Table 1 - Terminology. Table 2 - methods, tools, visualization packages. Methods for integrative data analysis of multi-omics data.

- Lee, Su-In, and Serafim Batzoglou. “Application of Independent Component Analysis to Microarrays.” Genome Biology 4, no. 11 (2003): R76. doi:10.1186/gb-2003-4-11-r76. - Independent Components Analysis theory and applications.

- Stein-O’Brien, Genevieve L., Raman Arora, Aedin C. Culhane, Alexander Favorov, Casey Greene, Loyal A. Goff, Yifeng Li, et al. “Enter the Matrix: Interpreting Unsupervised Feature Learning with Matrix Decomposition to Discover Hidden Knowledge in High-Throughput Omics Data.” BioRxiv, October 2, 2017. doi:10.1101/196915. - Matrix factorization and visualization. Refs to various types of MF methods.Terminology, Fig 1 explanation of MF in terms of gene expression and biological processes. References to biological examples.

- Libbrecht, Maxwell W., and William Stafford Noble. “Machine Learning Applications in Genetics and Genomics.” Nature Reviews. Genetics 16, no. 6 (June 2015): 321–32. doi:10.1038/nrg3920. - Machine learning in genomics. Supervised/unsupervised learning, semi-supervised, bayesian (incorporating prior knowledge), feature selection, imbalanced class sizes, missing data, networks. 

- Meng, Chen, Bernhard Kuster, Aedín C. Culhane, and Amin Moghaddas Gholami. “A Multivariate Approach to the Integration of Multi-Omics Datasets.” BMC Bioinformatics 15 (May 29, 2014): 162. doi:10.1186/1471-2105-15-162. - MCIA - multiple correspondence analysis for integrating multiple datasets. Statistics and implementation in `omicade4` - Multiple co-inertia analysis of omics datasets. https://bioconductor.org/packages/release/bioc/html/omicade4.html
