---
title: "Annotation"
author: "Mikhail Dozmorov"
date: "Fall 2017"
output:
  ioslides_presentation:
    css:
    - ../css/fonts.css
    - ../css/custom.css
    transition: faster
    widescreen: true
  beamer_presentation:
    colortheme: dolphin
    fig_caption: no
    fig_height: 6
    fig_width: 7
    fonttheme: structurebold
    theme: AnnArbor
---

## Gene identifiers {.larger}

**Gene**

- **Ensembl** ENSG00000139618
- **Entrez** Gene 675
- **Unigene** Hs.34012

**RNA transcript**

- **GenBank** BC026160.1
- **RefSeq** NM_000059
- **Ensembl** ENST00000380152

## HUGO Gene Nomenclature Committee (HGNC) Gene Names

This resource lists gene name synonyms, which is useful if you are conducting a comprehensive literature search and need to find articles about a gene that may have been called other names in the past.

https://www.genenames.org/

## Reference Sequences (RefSeq)

- Reference standard
    - Eukaryotes: genomic, transcript, protein sequences, derived from: 
        - Computation
        - Manual curation of submitted data
        - Collaboration with other experts

- > 69,000 organisms
    - >7000 viruses, >40,000 prokaryotes, >10,000 eukaryotes
 
http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4702849/

## RefSeq Accession Prefixes

NM_ = mRNA (experimentally supported)
XM_ = mRNA (predicted model)
NP_ = protein (experimentally supported)
XP_ = protein (predicted model)
NC_ = genomic/chromosome
NG_ = incomplete genomic assembly


https://www.ncbi.nlm.nih.gov/books/NBK21091/table/ch18.T.refseq_accession_numbers_and_mole/?report=objectonly

## NCBI Gene

https://www.ncbi.nlm.nih.gov/gene

<center><img src="img/ncbi_gene.png" height="370px" /></center>

## GeneCards

http://www.genecards.org/

<center><img src="img/genecards.png" height="370px" /></center>

## ID cross-mapping {.larger}

- There are many IDs
- Software tools recognize only a handful
- Humans better recognize gene names

## ID challenges {.larger}

- Avoid errors: map IDs correctly
    - Beware of 1-to-many mappings
- Gene name ambiguity â€“ not a good ID
    - e.g. FLJ92943, LFS1, TRP53, p53
    - Better to use the standard gene symbol, not aliases: TP53
- Excel error-introduction
    - OCT4 is changed to October-4  (open file/paste as text)
- Problems reaching 100% cross-mapping
    - E.g. due to version issues
    - Use multiple sources to increase coverage

## BiomaRt {.larger}

http://www.biomart.org/

https://bioconductor.org/packages/release/bioc/html/biomaRt.html

## BiomaRt {.larger}

The `getBM()` function has three arguments that need to be introduced: `filters`, `attributes` and `values`. 

- `Filters` define a restriction on the query. Tell BiomaRt what kind of IDs do you have, so it will look for it. The `listFilters()` function shows you all available filters in the selected dataset.
- `Attributes` define the values we are interested in to retrieve. Which IDs associated with your IDs you want to get. The `listAttributes()` function displays all available attributes in the selected dataset.
- `Values` is a vector of IDs you want to convert

## BiomaRt gotchas {.larger}

- `host` is the database version. For gene ID conversion, use the latest database.

For genomic coordinates, use database that corresponds to genome assembly version you are interested in 

## Other options {.larger}

Annotation data as R dataframes

R data package for annotating/converting Gene IDs 

- `annotables` R package by Stephen Turner, https://github.com/stephenturner/annotables

http://www.gettinggeneticsdone.com/2015/11/annotables-convert-gene-ids.html

